orchestration/flows/bl832/tomography_hpc.py

```diff
@@ -0,0 +1,343 @@
+
+        if self.client is None:
+            logger.error("NERSC client is required for job submission.")
+            raise ValueError("NERSC client is required for job submission.")
+
+        load_dotenv()
+
+        # Can't use this long term in production. Need to find a better way to handle credentials.
+        # Want to run this as the alsdev user
+        username = os.getenv("NERSC_USERNAME")
+        password = os.getenv("NERSC_PASSWORD")
+
+        user = self.client.user()
+
+        home_path = f"/global/homes/{user.name[0]}/{user.name}"
+        scratch_path = f"/pscratch/sd/{user.name[0]}/{user.name}"
+        logger.info(home_path)
+        logger.info(scratch_path)
+
+        image_name = "tomorecon_nersc_mpi_hdf5@sha256:cc098a2cfb6b1632ea872a202c66cb7566908da066fd8f8c123b92fa95c2a43c"
+        path = Path(file_path)
+        folder_name = path.parent.name
+        file_name = path.stem
+
+        job_script = f"""#!/bin/bash
+            #SBATCH -q preempt
+            #SBATCH -A als
+            #SBATCH -C cpu
+            #SBATCH --job-name=tomorecon_nersc_mpi_hdf5_1-0
+            #SBATCH --output={scratch_path}/nerscClient-test/%x_%j.out
+            #SBATCH --error={scratch_path}/nerscClient-test/%x_%j.err
+            #SBATCH -N 1
+            #SBATCH --ntasks-per-node 1
+            #SBATCH --cpus-per-task 64
+            #SBATCH --time=00:15:00
+            #SBATCH --exclusive
+
+            date
+
+            srun podman-hpc login registry.nersc.gov --username {username} --password {password}
+            srun podman-hpc run
+            --volume {home_path}/tomo_recon_repo/microct/legacy/sfapi_reconstruction.py:/alsuser/sfapi_reconstruction.py
+            --volume {home_path}/tomo_recon_repo/microct/legacy/input.txt:/alsuser/input.txt
+            --volume {scratch_path}/microctdata:/alsdata
+            --volume {scratch_path}/microctdata:/alsuser/ \
+            registry.nersc.gov/als/{image_name} \
+            python sfapi_reconstruction.py {file_name} {folder_name}
+
+            date
+            """
+
+        try:
+            logger.info("Submitting reconstruction job script to Perlmutter.")
+            job = self.client.perlmutter.submit_job(job_script)
+            job.complete()  # waits for job completion
+            logger.info("Reconstruction job completed successfully.")
+            return True
+        except Exception as e:
+            logger.error(f"Failed to submit or complete reconstruction job: {e}")
+            return False
+
+    def _submit_nersc_multi_resolution_job(
+        self,
+        file_path: str = None,
+    ) -> bool:
+        """Submit a multi-resolution tomography job to NERSC"""
+
+        if self.client is None:
+            logger.error("NERSC client is required for job submission.")
+            raise ValueError("NERSC client is required for job submission.")
+
+        load_dotenv()
+        username = os.getenv("NERSC_USERNAME")
```

Comment:
We really need to figure out how not to need user credentials for harbor.

Reply:
I noticed that our als project on Harbor is marked as 'private'. Perhaps we can set up another project for images we don't mind being 'public' that could get around needing credentials? I can open a ticket at NERSC to see what they suggest

![image](https://github.com/user-attachments/assets/0510acb3-40b3-42f6-955a-5bee20bb00ec)

Reply:
I see what you're saying, but we need to solve this somewhat globably across project.